When you change merc to a tank and leave the map editor it always gives a crash log

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues for each one!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong

In map editor you can add a tank to a map. First you need to mark some merc and change his character type. This situation is not causing a crash, but when you exit map editor after that, then the game is always giving a crash log with the same bug log.

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  Stable 0.20.0  | JA2 1.12 Gold ENG  |    no    |    Yes     |

[//]: # (Please also include:)
[//]: # Open map editor, change some merc to a tank, and then exit map editor.
[//]: # Windows 100

```
2023-11-13T18:33:02 [INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2023-11-13T18:33:03 [INFO] game\Init.cc: Beginning JA2 using -editor commandline argument...
2023-11-13T18:33:03 [INFO] game\GameScreen.cc: Aborting normal game mode and entering editor mode...
2023-11-13T18:33:03 [INFO] game\Editor\EditScreen.cc: Entering editor mode...
2023-11-13T18:33:25 [ERROR] sgp\SGP.cc: Game has been terminated due to an unrecoverable error: Invalid vehicle ID (class std::logic_error)
```
[//]: # (When pasting backtraces or any other logs with hashed enumerations, please format it as a code block.)
[//]: # (You can do that by selecting it and clicking the "<>" icon or manually putting three backticks [```] at each end.)

--- a/src/game/Tactical/Soldier_Control.cc
+++ b/src/game/Tactical/Soldier_Control.cc
@@ -6985,7 +6985,7 @@ void EVENT_SoldierBeginFirstAid( SOLDIERTYPE *pSoldier, INT16 sGridNo, UINT8 ubD
 void EVENT_SoldierEnterVehicle(SOLDIERTYPE& s, GridNo const gridno)
 {
 	SOLDIERTYPE const* const tgt = FindSoldier(gridno, FIND_SOLDIER_GRIDNO);
-	if (tgt && tgt->uiStatusFlags & SOLDIER_VEHICLE)
+	if (tgt && OK_ENTERABLE_VEHICLE(tgt))
 	{
 		VEHICLETYPE& v = GetVehicle(tgt->bVehicleID);
 		PutSoldierInVehicle(s, v);

--- a/src/game/Tactical/Soldier_Create.cc
+++ b/src/game/Tactical/Soldier_Create.cc
@@ -1,6 +1,7 @@
 #include "Soldier_Create.h"
 #include "ItemModel.h"
 #include "Overhead.h"
+#include "Soldier_Macros.h"
 #include "Soldier_Profile.h"
 #include "Animation_Control.h"
 #include "Animation_Data.h"
@@ -978,7 +979,7 @@ void InternalTacticalRemoveSoldier(SOLDIERTYPE& s, BOOLEAN const fRemoveVehicle)
 
 	if (s.ubScheduleID) DeleteSchedule(s.ubScheduleID);
 
-	if (s.uiStatusFlags & SOLDIER_VEHICLE && fRemoveVehicle)
+	if (EnterableVehicle(s) && fRemoveVehicle)
 	{
 		RemoveVehicleFromList(GetVehicle(s.bVehicleID));
 	}

--- a/src/game/Tactical/Soldier_Macros.h
+++ b/src/game/Tactical/Soldier_Macros.h
@@ -66,6 +66,14 @@ static inline BOOLEAN OkControllableMerc(const SOLDIERTYPE* const s)
 
 #define TANK(p)			((p)->ubBodyType == TANK_NE || (p)->ubBodyType == TANK_NW)
 
-#define OK_ENTERABLE_VEHICLE(p)	((p)->uiStatusFlags & SOLDIER_VEHICLE && !TANK((p)) && (p)->bLife >= OKLIFE)
+static inline bool EnterableVehicle(SOLDIERTYPE const& s)
+{
+	return (s.uiStatusFlags & SOLDIER_VEHICLE) && !TANK(&s);
+}
+
+static inline bool OK_ENTERABLE_VEHICLE(SOLDIERTYPE const * s)
+{
+	return EnterableVehicle(*s) && s->bLife >= OKLIFE;
+}
 
 #endif

--- a/src/game/Tactical/Vehicles.cc
+++ b/src/game/Tactical/Vehicles.cc
@@ -1,4 +1,3 @@
-#include "ItemModel.h"
 #include "LoadSaveVehicleType.h"
 #include "Map_Screen_Interface_Map.h"
 #include "SaveLoadGame.h"
@@ -787,8 +786,8 @@ void SetVehicleSectorValues(VEHICLETYPE& v, const SGPSector& sMap)
 void UpdateAllVehiclePassengersGridNo(SOLDIERTYPE* const vs)
 {
 	// If not a vehicle, ignore!
-	if (!(vs->uiStatusFlags & SOLDIER_VEHICLE)) return;
-	VEHICLETYPE const& v = pVehicleList[vs->bVehicleID];
+	if (!EnterableVehicle(*vs)) return;
+	VEHICLETYPE const& v = GetVehicle(vs->bVehicleID);
 
 	// Loop through passengers and update each guy's position
 	CFOR_EACH_PASSENGER(v, i)

